src/liballoc/lib.rs

```diff
@@ -163,7 +164,8 @@ mod boxed {
 #[cfg(test)]
 mod boxed_test;
 pub mod collections;
-#[cfg(target_has_atomic = "ptr")]
+#[cfg_attr(stage0, cfg(target_has_atomic = "ptr"))]
+#[cfg_attr(not(stage0), cfg(all(target_has_atomic = "ptr", target_has_atomic_cas)))]
```

Comment:
Instead of this `cfg_attr`, perhaps this could be:

```rust
#[cfg(any(
    all(stage0, target_has_atomic = "ptr"),
    all(not(stage0), target_has_atomic_cas, target_has_atomic = "ptr")
))]
```

That may be a bit easier to clean up after the next stage0

Reply:
Also, in terms of implementation, it may be easiest to piggy back on the existing `target_has_atomic` perhaps and do something like `target_has_atomic ="cas"` maybe?